packages/grafana-ui/src/types/datasource.ts

```diff
@@ -172,6 +172,11 @@ export abstract class DataSourceApi<
    */
   getQueryDisplayText?(query: TQuery): string;
 
+  /**
+   * Retrieve context for a given log row
+   */
+  getLogRowContext?(row: any, limit?: number): Promise<DataQueryResponse>;
```

Comment:
is a row completely untyped? don't we need the LogsModel and the row model is LogRowModel (and could be good to have both).

Reply:
Agree about LogRowModel. Not necessarily abut LogsModel - why would you need that?

Reply:
I am not 100% sure a single row can be enough information to construct a query for the context. You might need labels and the original request & response to create the context query.